OBSDOCS-206: Add multi ClusterLogForwarder docs and clean up existing documentation (#64935)
openshift/openshift-docs: modules/log-forwarding-modes.adoc

// Module included in the following assemblies:
//

> // * list of assemblies where this module is included

Reviewer: ??

:_content-type: CONCEPT
[id="log-forwarding-modes_{context}"]
= Log forwarding modes

There are two log forwarding modes available: legacy mode, and multi log forwarder mode.

[IMPORTANT]
====

> Only the Vector collector is supported for use with multi log forwarder mode. The Fluentd collector can only be used with the legacy `instance` log forwarder in the `openshift-logging` namespace.

Reviewer: > Only the Vector collector is supported for use with multi log forwarder mode.

Does this mean that other than Vector, no other collector can be used with this mode, or Vector cannot be used with legacy mode?

Reviewer: > used with the legacy `instance` log forwarder

Is this _legacy mode_?

Author: > Does this mean that other than Vector, no other collector can be used with this mode

Yes. The other option is the Fluentd collector (deprecated) and that can't be used with the multi log forwarder mode.

Author: Updated for clarity.

Reviewer 2: legacy mode:  The resources  clusterlogging  and ClusterLogForwarder must be deployed under openshift-logging namespace.  and the resource name must be instance.

New mode: 
1. The resources  clusterlogging  and ClusterLogForwarder can be deployed under any namespace.
2. ServiceAccount must be created and specified in ClusterLogForwarder.  
3. ServiceAccount must bind to logging roles(collect-application-logs,collect-audit-logs,                                                         
collect-infrastructure-logs )

====

[id="log-forwarding-modes-legacy_{context}"]
== Legacy mode

> In legacy mode, you can only use one log forwarder in your cluster. The `ClusterLogForwarder` resource in this mode must be named `instance`, and must be created in the `openshift-logging` namespace. The `ClusterLogForwarder` resource also requires a corresponding `ClusterLogging` resource, that also must be named `instance`, and must also be created in the `openshift-logging` namespace.

Reviewer: ```suggestion
In legacy mode, you can only use one log forwarder in your cluster. The `ClusterLogForwarder` resource in this mode must be named `instance`, and must be created in the `openshift-logging` namespace. The `ClusterLogForwarder` resource also requires a corresponding `ClusterLogging` resource named `instance` in the `openshift-logging` namespace.
```

[id="log-forwarding-modes-multi-clf_{context}"]
== Multi log forwarder mode

> Multi log forwarder mode is available in logging 5.8 and newer versions, and provides the following functionality:

Reviewer: ```suggestion
Multi log forwarder mode is available in logging 5.8 and later, and provides the following functionality:
```

* Administrators can control which users are allowed to define log collection and which logs they are allowed to collect.
* Users who have the required permissions are able to specify additional log collection configurations.

> * Cluster administrators who are migrating from the deprecated Fluentd collector to the Vector collector can deploy a new log forwarder separately from their existing deployment. The existing and new log forwarders can operate simultaneously while workloads are being migrated.

Reviewer: For ROSA and OSD, the cluster-admin role is not available. Users can have the dedicated-admin role.

Author: Updated to just "administrators"

> In multi log forwarder mode, you are not required to create a corresponding `ClusterLogging` resource for your `ClusterLogForwarder` resource. You can create multiple `ClusterLogForwarder` resources using any name, in any namespace, except one named `instance` in the `openshift-logging` namespace, because this is reserved for a log forwarder that supports the legacy workflow using the Fluentd collector.

Reviewer 3: User can't create CLF named `collector` in `openshift-logging` project either.

Author: Updated to add this, thanks

> In multi log forwarder mode, you are no longer required to create a corresponding `ClusterLogging` resource for your `ClusterLogForwarder` resource. You can create multiple `ClusterLogForwarder` resources using any name, in any namespace, except one named `instance` in the `openshift-logging` namespace, since this is reserved for a log forwarder that supports the legacy workflow using the Fluentd collector.

Reviewer: ```suggestion
In multi log forwarder mode, you are not required to create a corresponding `ClusterLogging` resource for your `ClusterLogForwarder` resource. You can create multiple `ClusterLogForwarder` resources using any name, in any namespace, except one named `instance` in the `openshift-logging` namespace, because this is reserved for a log forwarder that supports the legacy workflow using the Fluentd collector.
```
